lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -988,6 +988,188 @@ OpFoldResult ICmpOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+// Given a range of operands, computes the number of matching prefix and
+// suffix elements. This does not perform cross-element matching.
+static std::pair<size_t, size_t>
+computeCommonPrefixAndSuffix(const OperandRange &a, const OperandRange &b) {
+  size_t commonPrefixLength = 0;
+  size_t commonSuffixLength = 0;
+  size_t sizeA = a.size();
+  size_t sizeB = b.size();
+
+  for (; commonPrefixLength < std::min(sizeA, sizeB); commonPrefixLength++) {
+    if (a[commonPrefixLength] != b[commonPrefixLength]) {
+      break;
+    }
+  }
+
+  for (; commonSuffixLength < std::min(sizeA, sizeB) - commonPrefixLength;
+       commonSuffixLength++) {
+    if (a[sizeA - commonSuffixLength - 1] !=
+        b[sizeB - commonSuffixLength - 1]) {
+      break;
+    }
+  }
+
+  return {commonPrefixLength, commonSuffixLength};
+}
+
+static size_t getTotalWidth(const OperandRange &range) {
+  size_t totalWidth = 0;
+  for (auto operand : range) {
+    // getIntOrFloatBitWidth should never raise, since all arguments to ConcatOp
+    // are integers.
+    ssize_t width = operand.getType().getIntOrFloatBitWidth();
+    assert(width >= 0);
+    totalWidth += width;
+  }
+  return totalWidth;
+}
+
+static Optional<Value> findFirstNonEmptyValue(const OperandRange &range) {
+  for (auto op : range) {
+    if (op.getType().getIntOrFloatBitWidth() > 0) {
+      return mlir::Optional<Value>(op);
+    }
+  }
+
+  return None;
+}
+
+static bool isPredicateSigned(ICmpPredicate predicate) {
```

Comment:
Plz move this to being a static method on ICmpOp, similar to `ICmpOp::getFlippedPredicate`

Reply:
Done